core/CronArchive.php

```diff
@@ -904,7 +904,11 @@ public function isThereExistingValidPeriod(Parameters $params)
         $isYesterday = $params->getPeriod()->getLabel() == 'day' && $params->getPeriod()->getDateStart()->toString() == Date::factory('yesterday')->toString();
 
         $isPeriodIncludesToday = $params->getPeriod()->isDateInPeriod($today);
-        $minArchiveProcessedTime = $isPeriodIncludesToday ? Date::now()->subSeconds(Rules::getPeriodArchiveTimeToLiveDefault($params->getPeriod()->getLabel())) : null;
+        if (!$isPeriodIncludesToday) {
```

Comment:
@diosmosis can we maybe add a comment why we skip here?

Reply:
technically we could still keep the logic and it would just work the next time it runs after `minArchiveProcessedTime` has passed?

Reply:
@tsteur I left a comment w/ my reasoning here: https://github.com/matomo-org/matomo/pull/17379#issuecomment-806067847

I was thinking there could be weird edge cases where we rearchive say a month of data in the past, but only some of it gets archived. Eg, let's say we try to archive a specific funnel for the last month:

- user invalidates a day from a week ago (or a visit is tracked in the past for that day)
- archiver runs, the day archive from a week ago is rearchived (ts_archived = '2020-03-24 19:00:00')
- user changes a funnel, last month is scheduled to rearchive
- archiver runs, every day is invalidated except the day from a week ago. the data is archived but incorrect.

This isn't an issue I think for tracking in the past so we could keep it in that case, because if we don't invalidate, it's not removed from the "remember to invalidate" option values. So we'll do it again. But other types of invalidation it probably shouldn't be done.

Though only issue i can think of w/ this change for tracking in the past is if visits keep coming in, we keep invalidating, except we only do this once per site before archiving, so I don't think that will happen...

Reply:
> user changes a funnel, last month is scheduled to rearchive

do I get it right it wouldn't invalidate the day from a week ago because of a race condition if this happens shortly after the archiver ran? Technically, when scheduling to rearchive last month then it would also invalidate each day and week so they would be archived again?

Reply:
> do I get it right it wouldn't invalidate the day from a week ago because of a race condition if this happens shortly after the archiver ran?

In this specific example, yes. I suspect there are many strange edge cases like this that could happen.

> Technically, when scheduling to rearchive last month then it would also invalidate each day and week so they would be archived again?

I guess it's better to use an example of last2 months, where there's no period that contains today, and the new visit is tracked in the month before last.

It would invalidate the adjacent days, and the week. When the week gets archived, it would see there is a usable day archive and just use it since value = DONE_OK and ts_archived is recent and date does not include today (if my reasoning is correct).

Reply:
I think in some cases, we always want to invalidate, and in others we want to wait to make sure we're not invalidating to often. Tracking data in the past falls in the the second category. Rearchiving probably falls into the first...

Reply:
OK thanks 👍

Reply:
@tsteur took another look, and my reasoning for this was off, so removed the code

Reply:
👍

Reply:
we could merge the remaining changes then once tests pass?

Reply:
:+1: will do, working on test fixes now